lib/Models/GeoRssCatalogItem.js

```diff
@@ -0,0 +1,282 @@
+
+GeoRssCatalogItem.defaultUpdaters = clone(CatalogItem.defaultUpdaters);
+Object.freeze(GeoRssCatalogItem.defaultUpdaters);
+
+GeoRssCatalogItem.defaultSerializers = clone(CatalogItem.defaultSerializers);
+
+GeoRssCatalogItem.prototype._getValuesThatInfluenceLoad = function() {
+  return [this.url];
+};
+
+GeoRssCatalogItem.prototype._load = function() {
+  this._geoJsonItem = new GeoJsonCatalogItem(this.terria);
+  this._geoJsonItem.data = loadGeoRss(this);
+  this._geoJsonItem.name = this.name;
+  var that = this;
+
+  return this._geoJsonItem.load().then(function() {
+    if (!defined(that.rectangle)) {
+      that.rectangle = that._geoJsonItem.rectangle;
+    }
+  });
+};
+
+GeoRssCatalogItem.prototype._enable = function() {
+  if (defined(this._geoJsonItem)) {
+    this._geoJsonItem._enable();
+  }
+};
+
+GeoRssCatalogItem.prototype._disable = function() {
+  if (defined(this._geoJsonItem)) {
+    this._geoJsonItem._disable();
+  }
+};
+
+GeoRssCatalogItem.prototype._show = function() {
+  if (defined(this._geoJsonItem)) {
+    this._geoJsonItem._show();
+  }
+};
+
+GeoRssCatalogItem.prototype._hide = function() {
+  if (defined(this._geoJsonItem)) {
+    this._geoJsonItem._hide();
+  }
+};
+
+GeoRssCatalogItem.prototype.showOnSeparateMap = function(globeOrMap) {
+  if (defined(this._geoJsonItem)) {
+    return this._geoJsonItem.showOnSeparateMap(globeOrMap);
+  }
+};
+GeoRssCatalogItem.prototype.zoomTo = function() {
+  var that = this;
+  return when(this.load(), function() {
+    if (defined(that.nowViewingCatalogItem)) {
+      return that.nowViewingCatalogItem.zoomTo();
+    }
+
+    if (defined(that.rectangle)) {
+      return CatalogItem.prototype.zoomTo.call(that);
+    }
+
+    if (!defined(that.dataSource)) {
+      return;
+    }
+
+    return that.terria.currentViewer.zoomTo(that.dataSource);
+  });
+};
+
+function loadGeoRss(geoRssItem) {
+  return loadXML(proxyCatalogItemUrl(geoRssItem, geoRssItem.url)).then(function(
+    xml
+  ) {
+    if (typeof xml === "string") {
+      const parser = new DOMParser();
+      xml = parser.parseFromString(xml, "text/xml");
+    }
+    var documentElement = xml.documentElement;
+    var promise;
+    if (documentElement.localName === "rss") {
+      promise = geoRss2Convertor(xml);
+      resolveCatalogMetaData(
+        documentElement.getElementsByTagName("channel")[0],
+        geoRssItem
+      );
+    } else if (documentElement.localName === "feed") {
+      promise = geoRssAtomConvertor(xml);
+      resolveCatalogMetaData(documentElement, geoRssItem);
+    } else {
+      throw new RuntimeError("document is not valid");
+    }
+    return promise;
+  });
+}
+
+function resolveCatalogMetaData(documentElement, geoRssItem) {
+  var childNodes = documentElement.childNodes;
+
+  for (
+    var childNodeIndex = 0;
+    childNodeIndex < childNodes.length;
+    ++childNodeIndex
+  ) {
+    var child = childNodes[childNodeIndex];
+    if (child.nodeType !== 1 && child.localName === "item") {
```

Comment:
Should this line instead be the following?
```suggestion
    if (child.nodeType !== 1 || child.localName === "item") {
```

I don't think it will change the overall logic of the function as none of the if statements will be true anyway.
Also should this if statement also exclude "entry" from being processed?

Reply:
I missed that you added this as a suggestion, so I updated it on my end, an entry should be there and is now